src/Quahog/Client.php

```diff
@@ -257,14 +261,34 @@ private function _sendCommand($command)
     /**
      * A wrapper to cleanly read a response from clamd.
      *
+     * @param bool $removeId
+     * @param string $readUntil
      * @return string
+     * @throws ConnectionException
      */
-    private function _receiveResponse()
+    private function _receiveResponse($removeId = false, $readUntil = "\n")
     {
-        $result = $this->_socket->read(4096);
+        $result = null;
+        $readUntilLen = strlen($readUntil);
+        do {
+            if ($this->_socket->selectRead($this->_timeout)) {
+                $rt = $this->_socket->read(4096, PHP_NORMAL_READ);
+                if ($rt == "") {
+                    break;
+                }
+                $result .= $rt;
+                if (strcmp(substr($result, 0 - $readUntilLen), $readUntil) == 0) {
+                    break;
+                }
+            } else {
+                throw new ConnectionException("Timeout waiting to read response");
```

Comment:
I think throwing an exception is fine here - I'm not sure if it constitutes a breaking change though, as `_closeConnection()` does actually document that a `ConnectionException` could be thrown even though one never actually is (not sure why I did that!).

I'd vote that this change shouldn't mean a major version bump, but I'm open to other views of course :) @MassiveHiggsField what do you think?